### DISCUSSION TITLE
Automating Detector Selection and Parameter Fine-Tuning for Optimal Image Stitching

### DISCUSSION THREAD

**Discussion by fadishaar84** [OpenStitching/stitching#223] [Q&A] [answered]
Hello, thank you for sharing this excellent stitching tool. I have a question about using the --detector option sequentially. Is it possible to implement a functionality where the stitching algorithm attempts to use different detectors one after the other? For instance, if it fails to stitch all the provided images using SIFT, it would then try using ORB, and if that also fails, it would finally attempt using AKAZE. Can this be achieved? The reason for this is that some detectors may only stitch two of the three provided images, while others can stitch all of them. Additionally, if stitching fails, can the --confidence_threshold be adjusted accordingly, such as degreasing the value of --confidence_threshold parameter using different --detector values? How can this process be automated without needing to run the script manually for each detector and confidence threshold?  It would be great to add such a functionality to automate the process of fine-tuning the parameters until we achieve the best stitching result (considering that the best stitching algorithm is the one that can stitch all the provided images regardless of some minor distortions that might exist in the stitched image).

**Comment by lukasalexanderweber** [comment, marked as answer]
Hello @fadishaar84,

of course you can create a script to fine tune the image stitching process. 

```
from stitching import Stitcher
stitcher = Stitcher()
settings = {"detector": "sift", "confidence_threshold": 0.2}
stitcher = Stitcher(**settings)
panorama = stitcher.stitch(["img1.jpg", "img2.jpg", "img3.jpg"])
```

you can try different detectors and lower the `confidence_threshold` gradually till you get a good result.
